pyanaconda/core/configuration/anaconda.py

```diff
@@ -404,9 +408,30 @@ def set_from_opts(self, opts):
         # Set geolocation provider
         # FIXME: This will be removed once the boot option becomes a boolean
         if "geoloc" in opts and opts.geoloc and opts.geoloc != "0":
-            self.timezone._set_option("geolocation_provider", opts.geoloc)
+            self.timezone._set_option(
+                "geolocation_provider",
+                _convert_geoloc_provider_id_to_url(opts.geoloc)
+            )
 
         self.validate()
 
 
+def _convert_geoloc_provider_id_to_url(provider_id):
+    """Convert provider ID to URL of the corresponding service.
+
+    :param str provider_id: id of the geolocation provider service
+    :return str: URL to use
+    """
```

Comment:
If I'm reading it correctly. It would be valuable to have a note that this is only kept for backward compatibility. Or am I incorrect?

Reply:
Sorry, I missed this aspect, both back then and now. Yes, that's correct.

In fact, we should probably officially deprecate the possibility to use `inst.geoloc=<provider-id>` and ask to use `[Timezone] geolocation_provider = <url>` in conf. Both for RHEL 9 and Fedora 39.

We already shipped this in f37, but never added the deprecation notice to docs and release notes, it seems. I can't find any PR that would satisfy the TODO there: https://github.com/rhinstaller/anaconda/pull/4074#issue-1219112128